### PR TITLE
Adding VIA support for sixkeyboard

### PR DESCRIPTION
**Pull request by dayofdoom**
Adding support for the sixkeyboard by techkeys.us

## Description

Added a new folder called via to hold the new keymap for via.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR
None
* 

## Checklist

- [ ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [ ] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/sixkeyboard/keymaps/via/keymap.c [hunk 1]
```diff
@@ -0,0 +1,33 @@
+
+#include QMK_KEYBOARD_H
+#include "matrix.h"
```
Comment (drashna) [inline keyboards/sixkeyboard/keymaps/via/keymap.c:3]: This shouldn't be needed, actually. 
```suggestion
```

File: keyboards/sixkeyboard/keymaps/via/keymap.c [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    KC_TRNS, KC_TRNS, KC_TRNS, \
+    KC_TRNS, KC_TRNS, KC_TRNS  \
+  )
+};
+
+void matrix_scan_user(void) {
+  // jump to bootloaer when all keys are pressed
+  if (matrix_get_row(0) == 0b111 && matrix_get_row(1) == 0b111) {
+    clear_keyboard();
+    bootloader_jump();
```
Comment (drashna) [inline keyboards/sixkeyboard/keymaps/via/keymap.c:31]: This handles the clear keyboard and the like. 
```suggestion
    reset_keyboard();
```

https://github.com/qmk/qmk_firmware/blob/00fc35539d774c43cde383a8d2cc8b254787ce6c/quantum/quantum.c#L123-L145

**Comment by dayofdoom** [comment]
thanks for reviewing this. made the suggested changes. still new to qmk and keyboard firmware in general

File: keyboards/sixkeyboard/keymaps/via/keymap.c [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  ),
+
+  [2] = LAYOUT(
+    KC_TRNS, KC_TRNS, KC_TRNS, \
+    KC_TRNS, KC_TRNS, KC_TRNS  \
+  ),
+
+  [3] = LAYOUT(
+    KC_TRNS, KC_TRNS, KC_TRNS, \
+    KC_TRNS, KC_TRNS, KC_TRNS  \
```
Comment (fauxpark) [inline keyboards/sixkeyboard/keymaps/via/keymap.c:22]:
```suggestion
    KC_A, KC_B, KC_C,
    KC_D, KC_E, KC_F
  ),

  [1] = LAYOUT(
    KC_TRNS, KC_TRNS, KC_TRNS,
    KC_TRNS, KC_TRNS, KC_TRNS
  ),

  [2] = LAYOUT(
    KC_TRNS, KC_TRNS, KC_TRNS,
    KC_TRNS, KC_TRNS, KC_TRNS
  ),

  [3] = LAYOUT(
    KC_TRNS, KC_TRNS, KC_TRNS,
    KC_TRNS, KC_TRNS, KC_TRNS
```

File: keyboards/sixkeyboard/keymaps/via/keymap.c [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  )
+};
+
+void matrix_scan_user(void) {
+  // jump to bootloaer when all keys are pressed
+  if (matrix_get_row(0) == 0b111 && matrix_get_row(1) == 0b111) {
+    reset_keyboard();
+  }
+};
```
Comment (fauxpark) [inline keyboards/sixkeyboard/keymaps/via/keymap.c:31]:
```suggestion
```
Don't need this. `VIA_ENABLE = yes` implies `BOOTMAGIC_ENABLE = lite`, so you can hold the key at (0,0) in the matrix (in this case, the top left or "A" key) while plugging in the board to reset.

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by dayofdoom** [comment]
@fauxpark thanks for the suggestions.

**Review by fauxpark — APPROVED** [review]

File: keyboards/sixkeyboard/keymaps/via/keymap.c [hunk 5]
```diff
@@ -0,0 +1,24 @@
+
```
Comment (noroadsleft) [inline keyboards/sixkeyboard/keymaps/via/keymap.c:1]: Need a license header here.

You can find a sample in [`quantum/template/base/keyboard.c`](https://github.com/qmk/qmk_firmware/blob/f145696dad27893b11213b314797711541431a9f/quantum/template/base/keyboard.c).

Comment (dayofdoom) [reply keyboards/sixkeyboard/keymaps/via/keymap.c:16]: added license header. thank you

**Review by noroadsleft — APPROVED** [review]

**Comment by noroadsleft** [comment]
Thanks!

For future reference, we recommend against committing to your `master` branch as you've done here, because pull requests from modified `master` branches can make it more difficult to keep your QMK fork updated. It is highly recommended for QMK development – regardless of what is being done or where – to keep your master updated, but **NEVER** commit to it. Instead, do all your changes in a branch (branches are basically free in Git) and issue PRs from your branches when you're developing.

There are instructions on how to keep your fork updated here:

[**Best Practices: Your Fork's Master: Update Often, Commit Never**](https://docs.qmk.fm/#/newbs_git_using_your_master_branch)

[Fixing Your Branch](https://docs.qmk.fm/#/newbs_git_resynchronize_a_branch) will walk you through fixing up your `master` branch moving forward. If you need any help with this just ask.

**Comment by dayofdoom** [comment]
@noroadsleft Thanks for the helpful information and taking the time to help a newb.

**Pull request merged**